modules/library/metadata/src/main/java/org/geotools/util/factory/Hints.java

```diff
@@ -688,6 +688,11 @@
      */
     public static final ClassKey GEOMETRY_CLIP = new ClassKey("org.locationtech.jts.geom.Geometry");
 
+    /**
+     * Used when a rendering transformation is applied to a vector tile geometries, to lead the
+     * store using the clip mask to remove geometries outside tiles.
+     */
+    public static final Key REMOVE_BUFFER_PIXEL = new Key(Boolean.class);
```

Comment:
The name is unclear, there is no just a single "pixel", and in the future, things can be more complicated, like having a buffer in meters.
The implementation is also asymmetric, StreamingRenderer uses the clip from the feature metadata for tile rendering already, but here a new hint has been added to delegate to the store instead. Can we wrap the collection and clip just before passing the data to the rendering transformation instead? No need for new hints in this case.
If the rework is considered too big, at least let's call this "REMOVE_TILE_BUFFER" instead.

Reply:
fixed